packages/java-parser/src/index.js

```diff
@@ -62,12 +41,26 @@ function parse(inputText, entryPoint = "compilationUnit") {
     );
   }
 
+  if (lexResult.tokens.length === 0) {
+    const EOF = Object.assign({}, cst.children.EOF[0]);
+    cst.children.EOF = [EOF];
+    attachComments(cst.children.EOF, lexResult.groups.comments);
+  } else {
+    attachComments(lexResult.tokens, lexResult.groups.comments);
+  }
+
   return cst;
 }
 
 function attachComments(tokens, comments) {
   const attachComments = [...comments];
 
+  // edge case: when the file only contains comments
```

Comment:
is that the only case where there is only a single token?
What if there is only a single empty statement ";" won't that also be the case of single token file?

```java
;
```

Or is that not valid in java?

Reply:
Well the java compiler does not complains so it should be valid in Java I guess. I made changes to support single token.